genutil/common.go

```diff
@@ -454,3 +455,47 @@ func addNewChild(m map[string]*yang.Entry, k string, v *yang.Entry, errs []error
 	errs = append(errs, fmt.Errorf("%s was duplicate", v.Path()))
 	return errs
 }
+
+// TransformEntry makes changes to the given AST subtree returned by goyang
+// depending on the compress behaviour.
+// Currently, only PreferOperationalState entails a transformation, where
+// leafrefs pointing to config leaves are changed to point to state leaves.
+func TransformEntry(e *yang.Entry, compressBehaviour CompressBehaviour) util.Errors {
+	if compressBehaviour != PreferOperationalState {
+		return nil
+	}
+
+	pointLeafrefToState := func(e *yang.Entry) error {
```

Comment:
It might be worth adding a comment here indicating that in OpenConfig this is /always/ safe to do - because there must be a 'state' leaf for each 'config' leaf, and this is guaranteed by the linter. In other schemas this wouldn't necessarily be the case.